### PR TITLE
add dnd function

### PR DESCRIPTION
**Pull request by Sho-ki**
### Why

- Explain the Problem

### What

- Explain what you did

### Images

| Before 😭 | After 😃 |
| :-------: | :------: |

### CODE REVIEW

**Review by hiroshi-tsutsui — COMMENTED** [review]
can you remove all inline-css?
it's difficult to read.

File: src/types/workout.ts [hunk 1]
```diff
@@ -42,10 +53,20 @@ export enum Colors {
...
+  'black' = 'black',
 }
 
 export type CategoryColor = {
   category: Categories;
   color: Colors;
-}
+};
+
+export const dayCombination: { [key: string]: any } = {
```
Comment (hiroshi-tsutsui) [inline src/types/workout.ts:64]: any???

File: src/store/WokroutSetItemCxt.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Handlers } from '../components/WorkoutItems/hooks/useWorkoutItems';
+import { WorkoutSet } from '../types/workout';
+
+type CtxType = {
```
Comment (hiroshi-tsutsui) [inline src/store/WokroutSetItemCxt.tsx:5]: your declaring a type here so you don't need the word 'type' here.

File: src/store/WokroutSetItemCxt.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  orderChangedWeek: Array<WorkoutSet[]>;
+  dayOfToday: number;
+  onDeleteSetItem: (newArr: WorkoutSet[][], deleteId: number | null) => void;
```
Comment (hiroshi-tsutsui) [inline src/store/WokroutSetItemCxt.tsx:8]: `WorkoutSet[][]`
`Array<WorkoutSet[]>`
can you use one or the other??

File: src/components/WorkoutSets/index.tsx [hunk 4]
```diff
@@ -1,36 +1,210 @@
...
+  let dayOfWeek = getDate.getDay();
+
+  const [dayOfToday, setDayOfToday] = React.useState(dayOfWeek);
+
+  let week: Array<WorkoutSet[]> = [[], [], [], [], [], [], []];
+
+  const [orderChangedWeek, setOrderChangedWeek] = React.useState(week);
+  const [deleteIdList, setDeleteIdList] = React.useState<number[]>([]);
+
+  console.log(deleteIdList);
```
Comment (hiroshi-tsutsui) [inline src/components/WorkoutSets/index.tsx:39]: remove

File: src/components/WorkoutSets/index.tsx [hunk 5]
```diff
@@ -1,36 +1,210 @@
...
+    setOrderChangedWeek([...newWeekArr]);
+  }, []);
+
+  const onDragEnd = (result: DropResult) => {
+    const { destination, source, draggableId } = result;
+
+    if (!destination) {
+      return;
+    }
+    if (destination.droppableId === source.droppableId && destination.droppableId === 'workoutItems') {
```
Comment (hiroshi-tsutsui) [inline src/components/WorkoutSets/index.tsx:97]:
```suggestion
    if (source.droppableId === destination.droppableId && destination.droppableId === 'workoutItems') {
```

File: src/components/WorkoutSets/index.tsx [hunk 6]
```diff
@@ -1,36 +1,210 @@
...
+    }
+    setOrderChangedWeek([...newWeekArr]);
+  }, []);
+
+  const onDragEnd = (result: DropResult) => {
+    const { destination, source, draggableId } = result;
+
+    if (!destination) {
+      return;
+    }
```
Comment (hiroshi-tsutsui) [inline src/components/WorkoutSets/index.tsx:96]: I think one liner is more readable
```suggestion
    if (!destination) return
```

File: src/components/WorkoutItems/Items.tsx [hunk 7]
```diff
@@ -55,40 +51,64 @@ const WorkoutItems = ({ workoutitems, onClickEdit }: Props) => {
...
+                            variant='outlined'
+                            style={{
+                              borderRadius: 20,
+                              width: '80%',
+                              minHeight: '4em',
+                              border: `1px solid ${CategoryColors[workoutitem.color]}`,
+                              backgroundColor: `${CategoryColors[workoutitem.color]}`,
+                            }}
+                            onClick={() =>
+                              onClickEdit(
```
Comment (hiroshi-tsutsui) [inline src/components/WorkoutItems/Items.tsx:93]: type any is being used for onClickEdit on line 11.
Can you change that?

File: src/components/WorkoutDay/SetItem.tsx [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        key={`${workoutset.day_of_week}${idx}`}
+      >
+        {(provided) => (
+          <ListItem ref={provided.innerRef} {...provided.draggableProps}>
+            <div {...provided.dragHandleProps}>
+              <DragIndicatorIcon />
+            </div>
+            <button
+              disabled={isOnDeleteButton}
+              style={{}}
```
Comment (hiroshi-tsutsui) [inline src/components/WorkoutDay/SetItem.tsx:39]: is this necessary?

File: src/components/WorkoutDay/SetItem.tsx [hunk 9]
```diff
@@ -0,0 +1,124 @@
+import { ListItem, Typography } from '@material-ui/core';
+import { Draggable } from 'react-beautiful-dnd';
+import DragIndicatorIcon from '@mui/icons-material/DragIndicator';
```
Comment (hiroshi-tsutsui) [inline src/components/WorkoutDay/SetItem.tsx:3]: move all icons to `src/components/Icon/index.tsx`

**Review by hiroshi-tsutsui — COMMENTED** [review]
I'll check again onces this review has been changed

File: src/types/workout.ts (hunk 1 above)
Comment (hiroshi-tsutsui) [reply src/types/workout.ts:64]: and i think the key is number...(0, 1, 2, 3...)

File: src/components/WorkoutSets/index.tsx [hunk 10]
```diff
@@ -91,21 +124,15 @@ const WorkoutSets: React.FC<Props> = ({ workoutsets, workoutitems, categorycolor
...
-    if (!destination) {
-      return;
-    }
-    if (destination.droppableId === source.droppableId && destination.droppableId === 'workoutItems') {
-      return;
-    }
-    if (source.droppableId === 'workoutSetItems' && destination.droppableId === 'deleteList') {
-      return;
-    }
+    // If an invalid DnD is operated, return and nothing happens
```
Comment (hiroshi-tsutsui) [inline src/components/WorkoutSets/index.tsx:127]: this comment is good

File: src/pages/user/[userId]/edit.tsx [hunk 11]
```diff
@@ -1,14 +1,13 @@
...
-import { GetServerSideProps } from 'next';
-// import WorkoutSets from '../../components/WorkoutSets';
-import { resetServerContext } from 'react-beautiful-dnd';
 import dynamic from 'next/dynamic';
-const WorkoutSets = dynamic(import('../../../components/WorkoutSets'), { ssr: false });
+import { resetServerContext } from 'react-beautiful-dnd';
 
-// import WorkoutSets from '../../../components/WorkoutSets';
 import { CategoryColor } from '../../../types/workout';
+import { GetServerSideProps } from 'next';
```
Comment (hiroshi-tsutsui) [inline src/pages/user/[userId]/edit.tsx:6]: this is not self defined function

File: src/store/WokroutSetItemCxt.tsx [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
```
Comment (hiroshi-tsutsui) [inline src/store/WokroutSetItemCxt.tsx:1]: you only usually make one context per parent page.
I don't think this needs to be in a different file.

it's easier to understand where this context is coming from if you write it in the same file.

File: src/styles/globals.css [hunk 13]
```diff
@@ -14,3 +14,6 @@ a {
 * {
   box-sizing: border-box;
 }
+
+
+
```
Comment (hiroshi-tsutsui) [inline src/styles/globals.css:19]: けそか。

File: src/types/workout.ts [hunk 14]
```diff
@@ -1,13 +1,14 @@
 export type WorkoutSet = {
   id: number | null;
-  day_of_week: 'Sun' | 'Mon' | 'Tue' | 'Wed' | 'Thu' | 'Fri' | 'Sat' | null;
+  day_of_week: DayOfWeek;
```
Comment (hiroshi-tsutsui) [inline src/types/workout.ts:3]: good

File: src/components/WorkoutDay/index.tsx [hunk 15]
```diff
@@ -16,7 +16,7 @@ type Props = {
 
 const WorkoutDay: React.FC<Props> = ({ onPrevDayChangeHandler, onNextDayChangeHandler }: Props) => {
   const { orderChangedWeek, dayOfToday, saveSetItems } = React.useContext(WorkoutSetItemContext);
-
+  console.log(orderChangedWeek[dayOfToday]);
```
Comment (hiroshi-tsutsui) [inline src/components/WorkoutDay/index.tsx:19]: これはいらなそう

Comment (Sho-ki) [reply src/components/WorkoutDay/index.tsx]: I made a change
2b02ed485cd4d67c86fd75eff99cee7d5577399f

**Review by hiroshi-tsutsui — APPROVED** [review]
ちゃんと確認できてないけど、とりあえずLGTMってことで。

**Pull request merged**